lib/reversetunnel/track/tracker.go

```diff
@@ -52,268 +44,307 @@ func (c *Config) CheckAndSetDefaults() error {
 	if c.ProxyExpiry < 1 {
 		c.ProxyExpiry = DefaultProxyExpiry
 	}
-	if c.TickRate < 1 {
-		c.TickRate = 30 * time.Second
-	}
 	if c.ClusterName == "" {
 		return trace.BadParameter("missing ClusterName in track.Config")
 	}
 	return nil
 }
 
-// Tracker is a helper for tracking proxies located behind reverse tunnels
-// and triggering agent spawning as needed.  Tracker wraps a workpool.Pool
-// instance and manages a cache of proxies which *may* exist.  As proxies are
-// discovered, or old proxies expire, the target counts are automatically updated
-// for the associated address in the workpool.  Agents can attempt to "claim"
-// exclusivity for a given proxy, ensuring that multiple agents are not run
-// against the same proxy.
+// Tracker represents the view that a reverse tunnel client (i.e. an agentpool)
+// has over the reverse tunnel servers (i.e. proxies) that it's connected to and
+// that it knows about. Based on that information, the Tracker is in charge of
+// deciding if new connection attempts should be made, by giving out [Lease]s.
 type Tracker struct {
-	Config
-	mu     sync.Mutex
-	wp     *workpool.Pool
-	sets   *proxySet
-	cancel context.CancelFunc
+	proxyExpiry   time.Duration
+	clusterSuffix string
+
+	mu sync.Mutex
+
+	// connectionCount is nonpositive for full connectivity (agent mesh) mode, a
+	// positive number for the connection count of proxy peering mode.
+	connectionCount int
+
+	// cannotLease is a flag that should be reset whenever the tracker state is
+	// changed in such a way that it might be possible to grant leases.
+	cannotLease bool
```

Comment:
```suggestion
	// cannotLease being true indicates that no leases should be granted, but
	// false does not necessarily mean that leases *can* be granted.
	// this flag should be reset whenever the tracker state is
	// changed such that we *might* be able to grant leases.
	cannotLease bool
```

nit: didn't fully grasp the meaning of this field at first and was a bit thrown off by some of the places where this was being reset.  I *think* suggested comments might be a bit clearer, but idk. Minor nit, take it or leave it.